### PR TITLE
Adds more workbox docs

### LINKED ISSUE #1192: [Workbox content] Precaching do's and don'ts

### LINKED ISSUE #1198: [Workbox content] Troubleshooting and logging

### LINKED ISSUE #1200: [Workbox content] App Shell model for SPAs

### LINKED ISSUE #1201: [Workbox content] Navigation preload if HTML isn't cache first

### LINKED ISSUE #1204: [Workbox content] Precaching with Workbox

### LINKED ISSUE #1205: [Workbox content] Using Workbox without precaching

### LINKED ISSUE #1206: [Workbox content] Framework integration

### PR DESCRIPTION
**Pull request by malchata**
Issues addressed:
- Fixes #1192
- Fixes #1198
- Fixes #1200
- Fixes #1201
- Fixes #1204
- Fixes #1205
- Fixes #1206

Changes proposed in this pull request:

- Adds several more pages of Workbox documentation.

### CODE REVIEW

File: site/en/docs/workbox/app-shell-model/index.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+layout: "layouts/doc-post.njk"
+title: Service workers and the application shell model
+date: 2021-11-04
+description: >
+  How to pair the application shell model with a service worker in your SPA.
+---
+
+A common architectural feature of [single page web applications (SPA)](https://en.wikipedia.org/wiki/Single-page_application) is a minimal set of HTML, CSS, and JavaScript needed to power the global functionality of an application. In practice, this tends to be the header, navigation, and other common user interface elements that persist across all pages. When a service worker precaches this minimal UI's HTML and dependent assets, we call this the _application shell_.
+
+{% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/9nNaCUUYNSyBioL9eSiv.png", alt="A diagram of an application shell. It is a screenshot of a web page with a header at top, and a content area at the bottom. The header is labeled 'Application Shell', whereas the bottom is labeled as 'Content'", width="800", height="678" %}
```
Comment (rachelandrew) [inline site/en/docs/workbox/app-shell-model/index.md:11]:
```suggestion
{% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/9nNaCUUYNSyBioL9eSiv.png", alt="A diagram of an application shell. It is a screenshot of a web page with a header at top, and a content area at the bottom. The header is labeled 'Application Shell', whereas the bottom is labeled as 'Content'.", width="800", height="678" %}
```

File: site/en/docs/workbox/caching-strategies-overview/index.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ which is the URL for the network request currently being handled by the `fetch`
 which is the request method (e.g., `GET` or `POST`).
 - [`mode`](https://developer.mozilla.org/en-US/docs/Web/API/Request/mode),
```
Comment (rachelandrew) [inline site/en/docs/workbox/caching-strategies-overview/index.md]:
```suggestion
- [`mode`](https://developer.mozilla.org/docs/Web/API/Request/mode),
```

**Review by rachelandrew — APPROVED** [review]
There are a couple of places in here with TO DOs for linking stuff, which I assume you are aware of. We should check all the links before we go live anyway.

File: site/en/docs/workbox/caching-strategies-overview/index.md (hunk 2 above)
Comment (malchata) [reply site/en/docs/workbox/caching-strategies-overview/index.md]: Managed to fix this (and other offenders in an earlier doc) in a commit, so this is outdated currently.

**Pull request merged**